### PR TITLE
Improve Candid Connector Documentation

### LINKED ISSUE #5994: Write documentations for Candid Connector
**Description:**
This task is to write missing documentation and enhance the existing documentation of Candid connector.
- [x] Overview
- [x] Setup
- [x] Quick start
- [x] Examples
- [x] API Docs

**Describe your task(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by AzeemMuzammil**
## Purpose

$title.

Fixes: https://github.com/ballerina-platform/ballerina-library/issues/5994

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: build-config/resources/Ballerina.toml [hunk 1]
```diff
@@ -8,4 +8,4 @@ keywords = ["Candid", "Nonprofit Data", "Philanthropy Data", "Nonprofit APIs"]
 repository = "https://github.com/ballerina-platform/module-ballerinax-candid"
 icon = "icon.png"
 license = ["Apache-2.0"]
-distribution = "2201.8.0"
+distribution = "2201.8.4"
```
Comment (ayeshLK) [inline build-config/resources/Ballerina.toml]: Do we need to specify this distribution version ? Are we using any new lang features or OpenAPI tool features released under this particular version ?

File: examples/generate-charity-check-pdf/main.bal [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/os;
+import ballerinax/candid.charitycheckpdf;
+
+configurable string subscriptionKey = os:getEnv("SUBSCRIPTION_KEY");
```
Comment (ayeshLK) [inline examples/generate-charity-check-pdf/main.bal]: Since we have already mentioned in the Readme that the user need to configure this parameter in the `Config.toml` I think we do not need to check for environment variable.
```suggestion
configurable string subscriptionKey = ?;
```

File: examples/get-employer-information/main.bal [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/os;
+import ballerinax/candid.premier;
+
+configurable string subscriptionKey = os:getEnv("SUBSCRIPTION_KEY");
```
Comment (ayeshLK) [inline examples/get-employer-information/main.bal]: ditto

File: examples/search-essential-information/main.bal [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/os;
+import ballerinax/candid.essentials;
+
+configurable string subscriptionKey = os:getEnv("SUBSCRIPTION_KEY");
```
Comment (ayeshLK) [inline examples/search-essential-information/main.bal]: ditto

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -3,7 +3,7 @@ name: Publish Release
 on:
   workflow_dispatch:
   repository_dispatch:
-    types: [ connector-release-pipeline ]
+    types: [ stdlib-release-pipeline ]
```
Comment (NipunaRanasinghe) [inline .github/workflows/release.yml]: is this change intentional?

File: README.md [hunk 6]
```diff
@@ -1,80 +1,134 @@
...
 [![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/candid.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%2Fcandid)
 
 [Candid](https://candid.org/) is a non-profit organization that provides a comprehensive database of information about nonprofits, foundations, grantmakers, and philanthropists. Their mission is to connect people who want to change the world to the resources they need to do it.
 
-The `ballerinax/candid` module provides APIs to allow to access Candid's database that includes information on over 2 million nonprofits, including their mission, programs, finances, and leadership. And also provides information on over 100,000 foundations and grantmakers, as well as data on individual philanthropists.
+This Ballerina connector is designed to interface with Candid.org's API, enabling developers to access Candid's resources programmatically. It supports various operations, such as searching for nonprofit organizations, accessing grant data, and retrieving information about philanthropic trends and insights.
```
Comment (NipunaRanasinghe) [inline README.md]:
```suggestion
This Ballerina Candid connector is designed to interface with Candid.org's API, enabling developers to access Candid's resources programmatically. It supports various operations, such as searching for nonprofit organizations, accessing grant data, and retrieving information about philanthropic trends and insights.
```

File: ballerina/Module.md [hunk 7]
```diff
@@ -2,66 +2,119 @@
...
+essentials:V3EssentialsResponse|error result = essentials->/v3.post(query);
+```
+
+#### Premier API
 The following Ballerina program retrieves data on a nonprofit's financials, people, DEI, and IRS compliance validation for the given employer id number.
 
 ```ballerina
-import ballerinax/candid.premier;
+premier:V3PublicProfile|error result = check premier->/v3/["EMP-ID-NUM"];
+```
```
Comment (NipunaRanasinghe) [inline ballerina/Module.md:110]: We need to mention executing the `bal run` command as the last step of the quickstart guide.

File: ballerina/modules/charitycheckpdf/client.bal [hunk 8]
```diff
@@ -1,7 +1,4 @@
-// AUTO-GENERATED FILE. DO NOT MODIFY.
-// This file is auto-generated by the Ballerina OpenAPI tool.
```
Comment (NipunaRanasinghe) [inline ballerina/modules/charitycheckpdf/client.bal]: We need to keep this "AUTO-GENERATED" header part in all the generated files

File: examples/get-employer-information/main.bal [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+// Candid API Key Configurations
+premier:ApiKeysConfig apiKeyConfig = {
+    subscriptionKey: subscriptionKey
+};
+premier:Client premier = check new (apiKeyConfig);
+
+function getV3Premier() returns premier:V3PublicProfile|error {
+    premier:V3PublicProfile|error result = check premier->/v3/["EMP-ID-NUM"];
```
Comment (NipunaRanasinghe) [inline examples/get-employer-information/main.bal]:
```suggestion
    premier:V3PublicProfile profile = check premier->/v3/["EMP-ID-NUM"];
```

File: ballerina/modules/charitycheckpdf/client.bal (hunk 8 above)
Comment (AzeemMuzammil) [reply ballerina/modules/charitycheckpdf/client.bal]: This is getting removed when we generate, I think I might have just copied the license header last time, when we generate with the license header, the AUTO-GENERATED thing is getting removed and the license header is placed instead.

File: build-config/resources/Ballerina.toml (hunk 1 above)
Comment (AzeemMuzammil) [reply build-config/resources/Ballerina.toml]: Nope, we don't

File: .github/workflows/release.yml (hunk 5 above)
Comment (AzeemMuzammil) [reply .github/workflows/release.yml]: No, I think I got this from GitHub, will change,

File: ballerina/modules/charitycheckpdf/client.bal (hunk 8 above)
Comment (NipunaRanasinghe) [reply ballerina/modules/charitycheckpdf/client.bal]: @AzeemMuzammil yes thats the expected behaviour from the openAPI tool IINM (@lnash94 please confirm). 

As a solution for that we have to add the auto generated header part on top of the wso2 license text template file we keep in our resources

File: issue_template.md [hunk 10]
```diff
@@ -15,4 +15,4 @@
 
 
 **Related Issues:**
-<!-- Any related issues such as sub tasks, issues reported in other repositories (e.g component repositories), similar problems, etc. -->
\ No newline at end of file
+<!-- Any related issues such as sub tasks, issues reported in other repositories (e.g component repositories), similar problems, etc. -->
```
Comment (ayeshLK) [inline issue_template.md]: Do we need this file ? AFAIR all the issues related to connector modules are reported on Ballerina Library repository. Hence, we do not need issue-template here. Even issue tab should be disabled in this repo.

File: ballerina/modules/charitycheckpdf/client.bal (hunk 8 above)
Comment (AzeemMuzammil) [reply ballerina/modules/charitycheckpdf/client.bal]: I see, will do that,

File: .github/workflows/build-with-bal-test-graalvm.yml [hunk 11]
```diff
@@ -15,3 +13,5 @@ jobs:
     if: ${{ github.event_name != 'schedule' || (github.event_name == 'schedule' && github.repository_owner == 'ballerina-platform') }}
     uses: ballerina-platform/ballerina-standard-library/.github/workflows/build-with-bal-test-graalvm-connector-template.yml@main
     secrets: inherit
+    with:
+      additional-build-flags: "-x :candid-examples:build -Pgroups=mock"
```
Comment (Nuvindu) [inline .github/workflows/build-with-bal-test-graalvm.yml]:
```suggestion
      additional-build-flags: "-x :candid-examples:build -Pgroups=mock"

```

File: README.md [hunk 12]
```diff
@@ -1,80 +1,140 @@
 # Ballerina Candid Connector
 
-[![CI](https://github.com/ballerina-platform/module-ballerinax-candid/actions/workflows/ci.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-candid/actions/workflows/ci.yml)
+[![Build](https://github.com/ballerina-platform/module-ballerinax-candid/actions/workflows/ci.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-candid/actions/workflows/ci.yml)
+[![Trivy](https://github.com/ballerina-platform/module-ballerinax-candid/actions/workflows/trivy-scan.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-candid/actions/workflows/trivy-scan.yml)
 [![codecov](https://codecov.io/gh/ballerina-platform/module-ballerinax-candid/branch/master/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerinax-candid)
```
Comment (Nuvindu) [inline README.md]: I think we don't need the code coverage tag for connectors

File: README.md [hunk 13]
```diff
@@ -1,80 +1,140 @@
...
   - The Charity Check PDF API is a powerful tool that allows to download formatted Charity Check PDF reports for individual nonprofits.
-- Essentials API
+- **Essentials API**
   - The Essentials API is a lightweight API that provides basic information about nonprofits, such as their name, address, EIN, and mission statement. It is ideal for applications that need to quickly and easily identify and access basic nonprofit data.
-- Premier API
+- **Premier API**
   - The Premier API is a more comprehensive API that provides access to a wider range of nonprofit data, including financial information, staffing data, grantmaking data, and DEI data. It is ideal for applications that need to deep dive into the data of individual nonprofits or perform complex analyses of nonprofit data.
 
-## Samples
+## Setup Guide
```
Comment (Nuvindu) [inline README.md]: Can we use sentence case for the headings?

File: .github/workflows/ci.yml [hunk 14]
```diff
@@ -16,4 +16,4 @@ jobs:
     secrets: inherit
     with:
       repo-name: module-ballerinax-candid
-      additional-test-flags: -x test
+      additional-test-flags: -Pgroups=mock
```
Comment (Nuvindu) [inline .github/workflows/ci.yml]: We can use a single test suite for both mock and live test cases like here.
https://github.com/ballerina-platform/module-ballerinax-twilio/tree/master/ballerina/tests

In that way, we don't need to use `-Pgroups=mock` in `additional-test-flags`

File: docs/spec/sanitations.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
+# Introduction
```
Comment (Nuvindu) [inline docs/spec/sanitations.md]: Here, the topic should be similar to "Sanitations for OpenAPI specification"
And there should be meta information about authors, reviewers, etc.

File: ballerina/Package.md [hunk 16]
```diff
@@ -1,67 +1,135 @@
...
+
+The `Candid` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/ballerina-platform/module-ballerinax-candid/tree/main/examples) to understand how to interact with the Candid.org API for tasks such as generating compliance reports, searching for essential nonprofit information, and retrieving detailed employer data. 
+
+1. [Generate Charity Check PDF](https://github.com/ballerina-platform/module-ballerinax-candid/tree/main/examples/generate-charity-check-pdf) - Generate a detailed Charity Check PDF for a specified nonprofit organization using Candid.org's API.
+
+2. [Search Essential Information](https://github.com/ballerina-platform/module-ballerinax-candid/tree/main/examples/search-essential-information) - Search for and retrieve essential information about nonprofit organizations through the Candid.org Essentials API.
+
+3. [Get Employer Information](https://github.com/ballerina-platform/module-ballerinax-candid/tree/main/examples/get-employer-information) - Obtain comprehensive employer information for nonprofit organizations using the Candid.org Premier API.
+
+## Report Issues
```
Comment (Nuvindu) [inline ballerina/Package.md]: Do we need this `Report issues` and `Useful links` topics here?

File: ballerina/modules/charitycheckpdf/tests/README.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
+# Running Tests
+
+There are two test environments for the Candid connector. The default test environment is the mock server for Candid API. The other test environment is the actual Candid API. You can run the tests in either of these environments.
+
+## Running Tests in the Mock Server
```
Comment (Nuvindu) [inline ballerina/modules/charitycheckpdf/tests/README.md]: Use sentence case here as well. Check for other places

File: ballerina/modules/premier/tests/test.bal [hunk 18]
```diff
@@ -17,19 +14,39 @@
...
+@test:BeforeSuite
+function initializeClient() returns error? {
+    if isTestOnLiveServer {
+        log:printInfo("Initializing client for Candid server");
+        premier = check new (apiKeyConfig, serviceUrl = "https://api.candid.org/premier");
+    } else {
+        log:printInfo("Initializing client for mock server");
+        premier = check new (
+            apiKeyConfig = {
+                subscriptionKey: "6006e88b7fc2e0c31fbcb744cca10cafa280341758cd1db45fc1b29b05305dc0"
```
Comment (Nuvindu) [inline ballerina/modules/premier/tests/test.bal]: It would be better to use a configurable variable for `subscriptionKey`

File: examples/search-essential-information/Search essential information.md [hunk 19]
```diff
@@ -0,0 +1,24 @@
+# Search Essential Information
```
Comment (Nuvindu) [inline examples/search-essential-information/Search essential information.md]: Can we use sentence case in example docs as well?

File: examples/generate-charity-check-pdf/Generate charity check pdf.md [hunk 20]
```diff
@@ -0,0 +1,24 @@
+# Generate Charity Check PDF
+
+This example demonstrates how to generate a Charity Check PDF using the Candid.org API. It focuses on creating a comprehensive report for a specified nonprofit organization, including its compliance status, financial information, and charitable validation. The process involves making an API call to the CharityCheckPDF API, specifying the nonprofit's identifier, and receiving a PDF report that can be downloaded.
```
Comment (Nuvindu) [inline examples/generate-charity-check-pdf/Generate charity check pdf.md]:
```suggestion
This example demonstrates how to generate a Charity Check PDF using the Candid.org API. It focuses on creating a comprehensive report for a specified nonprofit organization, including its compliance status, financial information, and charitable validation. The process involves making an API call to the Charity Check PDF API, specifying the nonprofit's identifier, and receiving a PDF report that can be downloaded.
```

File: ballerina/Package.md (hunk 16 above)
Comment (AzeemMuzammil) [reply ballerina/Package.md]: Was following this, 
https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/blob/master/ballerina/Package.md#report-issues

File: examples/search-essential-information/Search essential information.md (hunk 19 above)
Comment (AzeemMuzammil) [reply examples/search-essential-information/Search essential information.md]: Are we using sentence cases for titles as well?

Comment (Nuvindu) [reply examples/search-essential-information/Search essential information.md]: Yeah, it seems so
https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/tree/master/examples/process-mails

Comment (AzeemMuzammil) [reply examples/search-essential-information/Search essential information.md]: and check this, https://github.com/ballerina-platform/module-ballerinax-googleapis.gmail/blob/master/examples/reply-mails/Automated%20Email%20Responses.md :)
Shall we confirm this?

Comment (AzeemMuzammil) [reply examples/search-essential-information/Search essential information.md:1]: But in most cases, it is using sentence case, but it is not conventional to use sentence case for a title, will verify this

Comment (AzeemMuzammil) [reply examples/search-essential-information/Search essential information.md]: will, go with sentence case

Comment (Nuvindu) [reply examples/search-essential-information/Search essential information.md]: Yeah. @NipunaRanasinghe what would be the better way here?

File: ballerina/Package.md (hunk 16 above)
Comment (Nuvindu) [reply ballerina/Package.md]: I think we don't need this. @niveathika what's your opinion on this?

Comment (niveathika) [reply ballerina/Package.md]: This is our old document structure and I kept it as it is.  since the Package.md file is not used anywhere for documentation display, +1 to remove

File: examples/search-essential-information/Search essential information.md (hunk 19 above)
Comment (niveathika) [reply examples/search-essential-information/Search essential information.md]: We should follow sentence case as it is the recommendation from our docs team. I will update gmail examples references.

**Review by NipunaRanasinghe — APPROVED** [review]
